scripts/helper.sh

```diff
@@ -38,3 +38,67 @@ function fail_if_error() {
     exit $retVal
   fi
 }
+
+function check_image_artifact_exists() {
+  local FILE=$1
+  if [[ ! -f "$FILE" ]]; then
+    echo "Error: Image artifact $FILE does not exist."
+    set_job_status "error"
+    exit 1
+  fi
+}
+
+# Set up the resolv.conf file for the image root
+# This is used to ensure that the image root can resolve DNS names during customization.
+function setup_resolv() {
+    # Copy the container's /etc/resolv.conf to the image root for customization
+    local IMAGE_ROOT_DIR=$1
+    local IMAGE_ROOT_RESOLV=$IMAGE_ROOT_DIR/etc/resolv.conf
+    local TMP_RESOLV=$IMAGE_ROOT_DIR/tmp/resolv.conf
+    if [ -f "$IMAGE_ROOT_RESOLV" -o -L "$IMAGE_ROOT_RESOLV" ]; then
+        mv "$IMAGE_ROOT_RESOLV" "$TMP_RESOLV"
+    fi
+    cp --remove-destination /etc/resolv.conf "$IMAGE_ROOT_RESOLV"
```

Comment:
If the output from this script will end up in a pod log, consider running commands with the verbose flag, for improved debug-ability. Obviously this won't work if this script needs to produce no output, or if these functions are called so many times that it would produce a lot of noise. Just wanted to mention it as a suggestion. Feel free to dismiss it if appropriate.

This applies to several other commands in this script, but I'll just make the suggestion here, and you can do as you think is wise.

```suggestion
    cp -v --remove-destination /etc/resolv.conf "$IMAGE_ROOT_RESOLV"
```